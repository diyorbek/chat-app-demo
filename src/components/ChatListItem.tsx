import {
  ListItem,
  ListItemButton,
  ListItemAvatar,
  Avatar,
  ListItemText,
  Divider,
  Typography,
} from '@mui/material';

interface Props {
  label: string;
  avatarUrl?: string;
  isActive?: boolean;
  isHighlighted?: boolean;
  isLast?: boolean;
  onClick?: () => void;
}

export function ChatListItem({
  label,
  avatarUrl,
  isActive,
  isHighlighted,
  isLast,
  onClick,
}: Props) {
  return (
    <>
      <ListItem
        aria-label={`item ${label}`}
        alignItems="flex-start"
        disablePadding
        secondaryAction={
          isHighlighted && (
            <Typography
              fontSize={36}
              color="orange"
              style={{ userSelect: 'none' }}
            >
              &bull;
            </Typography>
          )
        }
      >
        <ListItemButton selected={isActive} onClick={onClick}>
          <ListItemAvatar>
            <Avatar alt={label} src={avatarUrl} />
          </ListItemAvatar>
          <ListItemText
            primary={label}
            primaryTypographyProps={{
              fontWeight: isHighlighted ? 'bold' : undefined,
            }}
          />
        </ListItemButton>
      </ListItem>

      {!isLast && <Divider component="li" variant="inset" />}
    </>
  );
}
