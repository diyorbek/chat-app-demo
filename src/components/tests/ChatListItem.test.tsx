import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ChatListItem } from '../ChatListItem';

it('renders chat list item correctly', () => {
  render(<ChatListItem label="Jonh" />);

  expect(screen.getByLabelText('item Jonh')).toHaveTextContent('Jonh');
  expect(screen.getByRole('button')).toBeEnabled();
});
