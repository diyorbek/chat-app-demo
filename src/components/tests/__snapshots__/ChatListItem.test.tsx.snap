// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders chat list item correctly 1`] = `
<div>
  <li
    aria-label="item"
    class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart css-dwve99-MuiListItem-root"
  >
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Jonh
        </span>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </li>
  <li
    class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
    role="separator"
  />
</div>
`;
